### ISSUE TITLE
EstimatedTotalReadoutTime vs. TotalReadoutTime

### ISSUE THREAD

**Issue by arokem** [PennLINC/qsiprep#453]
I am processing a (Phillips) dataset that, after conversion from dicoms using dcm2niix has an `EstimatedTotalReadoutTime` field, but no `TotalReadoutTime` key in the json sidecars. 

This raises an error right around here: https://github.com/PennLINC/qsiprep/blob/master/qsiprep/interfaces/epi_fmap.py#L202, Because: https://github.com/PennLINC/qsiprep/blob/master/qsiprep/interfaces/epi_fmap.py#L17

First of all, does anyone know what the dangers are of using an *estimated* readout time, in the absence of other information? Also, would it make sense to also accept the estimated value here (with/without) additional warning?

For now, I'll go ahead and patch the estimated value in as the "precise" value and see what happens. I can report back about my findings.

**Comment by mattcieslak** [comment]
Hi Ariel, using `EstimatedTotalReadoutTime` should be fine. But of course please let me know if you find out differently. Is the value the same for both phase encoding directions?

**Comment by arokem** [comment]
Yes. Both PE directions have the same value of `EstimatedTotalReadoutTime`. 

Interestingly, this data also didn't include a `PhaseEncodingDirection` field in the side-car, so I had to patch that in too... 

Would it make sense to add some logic to the code to fall back to the `EstimatedTotalReadoutTime` value when a `TotalReadoutTime` field is not available?

**Comment by arokem** [comment]
And it's still running, but I will let you know when it's done and when I have a chance to take a close look how things went.

**Comment by cookpa** [comment]
https://fsl.fmrib.ox.ac.uk/fsl/fslwiki/eddy/Faq#Does_it_matter_what_I_put_into_my_--acqp_file.3F

> Does it matter what I put into my --acqp file?
> 
> Almost not at all, as long as the same file is used for both [topup](https://fsl.fmrib.ox.ac.uk/fsl/fslwiki/topup) and eddy.

I've seen users on the FSL users list cautioned not to use extremely wrong values (like orders of magnitude away from the usual reality) but a default in the vicinity of 0.05 seems to be safe if you are using topup with reversed PE field maps with unknown but equal readout time and opposite phase encoding directions, followed by eddy.

**Comment by arokem** [comment]
Thanks @cookpa: it sounds to me like what you are suggesting is that if an "estimated" value is accepted, some sanity checks should also be introduced to make sure that things are not far out of spec?

**Comment by cookpa** [comment]
I think an estimated value in the BIDS sidecar from `dcm2niix` would be trustworthy. I just remember some people who didn't know the readout times writing `acqparams.txt` by hand and putting things like "1.0". 

I found the issue where it's explained why Philips scans have "Estimated" total readout time: https://github.com/rordenlab/dcm2niix/issues/377

**Comment by arokem** [comment]
Thanks for unearthing that! I read [this](https://github.com/rordenlab/dcm2niix/issues/377#issuecomment-603905172) comment to say that it would be reasonable to implement a heuristic approach in qsiprep, where: 

1. In the absence of a `TotalReadoutTime` field, the software looks to see whether an `EstimatedTotalReadoutTime` field exists
2. Comparison between the two paired images is done to verify that the values are identical (similar?)
3. (optional?) A sanity check is conducted on the order of magnitude of these values to make sure that it's not too far from reasonable.
4. (optional?) raise a warning. It's unlikely that anyone would see this warning given how verbose the software is 😄 
5. (optional?) add a `--allow-estimated-readout` flag to make users explicitly ask for this to happen.

WDYT?
